sdk/python/feast/infra/aws.py

```diff
@@ -1,23 +1,165 @@
+            print("Deploying feature server...")
+
+            assert self.repo_config.repo_path
+            with open(self.repo_config.repo_path / "feature_store.yaml", "rb") as f:
+                config_bytes = f.read()
+                config_base64 = base64.b64encode(config_bytes).decode()
+
+            resource_name = self._get_lambda_name()
+            lambda_client = boto3.client("lambda")
+            api_gateway_client = boto3.client("apigatewayv2")
+            function = aws_utils.get_lambda_function(lambda_client, resource_name)
+
+            if function is None:
+                # If the Lambda function does not exist, create it.
+                print("  Creating AWS Lambda...")
+                lambda_client.create_function(
+                    FunctionName=resource_name,
+                    Role=self.repo_config.feature_server.execution_role_name,
+                    Code={"ImageUri": image_uri},
+                    PackageType="Image",
+                    MemorySize=1769,
+                    Environment={
+                        "Variables": {
+                            SERVER_CONFIG_BASE64_ENV_NAME: config_base64,
+                            FEAST_USAGE: "False",
+                        }
+                    },
+                )
+                function = aws_utils.get_lambda_function(lambda_client, resource_name)
+                assert function
+            else:
+                # If the feature_store.yaml has changed, need to update the environment variable.
+                env = function.get("Environment", {}).get("Variables", {})
+                if env.get(SERVER_CONFIG_BASE64_ENV_NAME) != config_base64:
+                    # Note, that this does not update Lambda gracefully (e.g. no rolling deployment).
+                    # It's expected that feature_store.yaml is not regularly updated while the lambda
+                    # is serving production traffic. However, the update in registry (e.g. modifying
+                    # feature views, feature services, and other definitions does not update lambda).
+                    print("  Updating AWS Lambda...")
+
+                    lambda_client.update_function_configuration(
+                        FunctionName=resource_name,
+                        Environment={
+                            "Variables": {SERVER_CONFIG_BASE64_ENV_NAME: config_base64}
+                        },
+                    )
+
+            api = aws_utils.get_first_api_gateway(api_gateway_client, resource_name)
+            if not api:
+                # If the API Gateway doesn't exist, create it
+                print("  Creating AWS API Gateway...")
+                api = api_gateway_client.create_api(
+                    Name=resource_name,
+                    ProtocolType="HTTP",
+                    Target=function["FunctionArn"],
+                    RouteKey="POST /get-online-features",
+                )
+                assert api
```

Comment:
super nitty, we should raise modeled exceptions for these things instead of `AssertionErrors`.